src/MainView.vala

```diff
@@ -68,22 +68,20 @@ public class DateTime.MainView : Gtk.Grid {
             xalign = 1
         };
 
-        auto_time_zone_icon = new Gtk.Image ();
-        auto_time_zone_icon.gicon = new ThemedIcon ("location-inactive-symbolic");
-        auto_time_zone_icon.pixel_size = 16;
+        auto_time_zone_icon = new Gtk.Image () {
+            gicon = new ThemedIcon ("location-inactive-symbolic"),
+            pixel_size = 16
+        };
 
-        var auto_time_zone_icon_context = auto_time_zone_icon.get_style_context ();
-        auto_time_zone_icon_context.add_class ("auto-timezone-label");
-        auto_time_zone_icon_context.add_provider (css_provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION);
+        weak Gtk.StyleContext auto_time_zone_icon_context = auto_time_zone_icon.get_style_context ();
+        auto_time_zone_icon_context.add_class ("accent");
+        auto_time_zone_icon_context.add_class ("purple");
```

Comment:
Instead of creating an unnecessary var for this, let's do it directly for more readability.

```suggestion
        auto_time_zone_icon.get_style_context ().add_class ("accent");
        auto_time_zone_icon.get_style_context ().add_class ("purple");
```

Reply:
Well it's not unnecessary because now you're getting it twice instead of once. So this is not a good suggestion

Reply:
| Getting it Twice | Calling it Once via New Var |
| :--- | :--- |
| No vars created thus no more valac internal C code bloat, more optimized | Creates a new var (more memory space used) for just this one job to be discarded, not optimized |

Things like this are more apparent on strained hardware like a Pi.

Reply:
Dude, if you want to get really technical. You're arguing over using 32 **bits** of memory for storing a pointer instead of calling a function twice. Calling a function has to push and pop a whole call stack and involves several machine code instructions.

I know for a fact that getting it twice is going to be way less efficient. But we're talking microseconds or even nanoseconds of difference here and bits of memory, so it's not worth arguing over.

Reply:
> You're arguing over using 32 **bits** of memory for storing a pointer instead of calling a function twice. Calling a function has to push and pop a whole call stack and involves several machine code instructions.

Sure, in this instance it's just one, but imagine if the code has it everywhere. Push & pop and several machine code instructions is way less of a problem because CPUs are designed for that, than flooding the memory stack with one-off vars.

> it's not worth arguing over.

Not my intention. I guess text really doesn't convey emotions.